Sources/Web3Core/EthereumABI/ABIElements.swift

```diff
@@ -432,61 +432,63 @@ extension ABI.Element.Function {
 
 extension ABI.Element.Constructor {
     public func decodeInputData(_ rawData: Data) -> [String: Any]? {
-        return Web3Core.decodeInputData(rawData, inputs: inputs)
+        return ABI.Element.decodeInputData(rawData, inputs: inputs)
     }
 }
 
-/// Generic input decoding function.
-/// - Parameters:
-///   - rawData: data to decode. Must match the following criteria: `data.count == 0 || data.count % 32 == 4`.
-///   - methodEncoding: 4 bytes representing method signature like `0xFFffFFff`. Can be omitted to avoid checking method encoding.
-///   - inputs: expected input types. Order must be the same as in function declaration.
-/// - Returns: decoded dictionary of input arguments mapped to their indices and arguments' names if these are not empty.
-/// If decoding of at least one argument fails, `rawData` size is invalid or `methodEncoding` doesn't match - `nil` is returned.
-private func decodeInputData(_ rawData: Data,
-                                 methodEncoding: Data? = nil,
-                                 inputs: [ABI.Element.InOut]) -> [String: Any]? {
-    let data: Data
-    let sig: Data?
-
-    switch rawData.count % 32 {
-    case 0:
-        sig = nil
-        data = Data()
-        break
-    case 4:
-        sig = rawData[0 ..< 4]
-        data = Data(rawData[4 ..< rawData.count])
-    default:
-        return nil
-    }
+extension ABI.Element {
```

Comment:
I'd make it part of the ABIDecoder instead.

Reply:
It would be required to make it `internal` then. Is it ok?

Reply:
Anyway, I've made it.